components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts

```diff
@@ -171,6 +173,35 @@ handler.on(PanelActions.signMessageProcessed.getType(), async (store, payload: S
   apiProxy.closeUI()
 })
 
+handler.on(PanelActions.signHardwareMessage.getType(), async (store, messageData: SignMessageData) => {
+  const apiProxy = await getAPIProxy()
+  const braveWalletService = apiProxy.braveWalletService
+  const result = await apiProxy.walletHandler.getWalletInfo()
+  const hardwareAccount = findHardwareAccountInfo(result.accountInfos, messageData.address)
+  if (hardwareAccount && hardwareAccount.hardware) {
+    let deviceKeyring = await apiProxy.getKeyringsByType(hardwareAccount.hardware.vendor)
+    deviceKeyring.signPersonalMessage(hardwareAccount.hardware.path, hardwareAccount.address, messageData.message).
+      then(async (signature: string) => {
+        await braveWalletService.notifyHardwareSignMessageRequestProcessed(true, messageData.id, signature, '')
+        apiProxy.closeUI()
+      }).catch(async (error: any) => {
+        await braveWalletService.notifyHardwareSignMessageRequestProcessed(false, messageData.id, '', error.message)
+        apiProxy.closeUI()
```

Comment:
Why don't we call `PanelActions.signHardwareMessageProcessed` directly here
so we don't have to do https://github.com/brave/brave-core/pull/10568/files#r734093561